Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+///  `BTAmericanExpressClient` enables you to look up the rewards balance of American Express cards.
+@objcMembers public class BTAmericanExpressClient: NSObject {
+    
+    private var apiClient: BTAPIClient
+    
+    ///  Creates an American Express client.
+    /// - Parameter apiClient: An instance of `BTAPIClient`
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+    
+    ///  Gets the rewards balance associated with a Braintree nonce. Only for American Express cards.
+    /// - Parameters:
+    ///   - nonce: A nonce representing a card that will be used to look up the rewards balance.
+    ///   - currencyIsoCode:  The currencyIsoCode to use. Example: 'USD'
+    ///   - completion:  A completion block that is invoked when the request has completed. If the request succeeds,
+    ///   `rewardsBalance` will contain information about the rewards balance and `error` will be `nil` (see exceptions in note);
+    ///   if it fails, `rewardsBalance` will be `nil` and `error` will describe the failure.
+    ///  - Note: If the nonce is associated with an ineligible card or a card with insufficient points, the rewardsBalance will contain this information as `errorMessage` and `errorCode`.
+    public func getRewardsBalance(forNonce nonce: String, currencyIsoCode: String, completion: @escaping (BTAmericanExpressRewardsBalance?, Error?) -> Void) {
+    
+        let parameters = ["currencyIsoCode": currencyIsoCode,
+                          "paymentMethodNonce": nonce]
+        self.apiClient.sendAnalyticsEvent("ios.amex.rewards-balance.start")
```

Comment:
```suggestion
        apiClient.sendAnalyticsEvent("ios.amex.rewards-balance.start")
```
Small nit: can we commit `self` where it isn't needed to improve readability?

Reply:
Good eye - we can remove this in a lot of places